R/FilterStateRange.R

```diff
@@ -39,30 +39,46 @@ RangeFilterState <- R6::R6Class( # nolint
     #' \item{`"matrix"`}{ `varname` in the condition call will be returned as `<input_dataname>[, <varname>]`}
     #' }
     initialize = function(x,
+                          x_filtered,
                           varname,
                           varlabel = character(0),
                           input_dataname = NULL,
                           extract_type = character(0)) {
       stopifnot(is.numeric(x))
       stopifnot(any(is.finite(x)))
 
-      super$initialize(x, varname, varlabel, input_dataname, extract_type)
+      #validation on x_filtered here
+      super$initialize(x, x_filtered, varname, varlabel, input_dataname, extract_type)
       var_range <- range(x, finite = TRUE)
 
+      private$inf_filtered_count <- reactive(sum(is.infinite(x_filtered())))
+
       private$set_choices(var_range)
       self$set_selected(var_range)
 
-      private$histogram_data <- if (sum(is.finite(x)) >= 2) {
-        as.data.frame(
-          stats::density(x, na.rm = TRUE, n = 100)[c("x", "y")] # 100 bins only
-        )
-      } else {
-        data.frame(x = NA_real_, y = NA_real_)
-      }
+      private$data_count <- length(x)
+
       private$inf_count <- sum(is.infinite(x))
       private$is_integer <- checkmate::test_integerish(x)
       private$keep_inf <- reactiveVal(FALSE)
 
+      pretty_range_inputs <- private$get_pretty_range_inputs(x)
+
+      private$unfiltered_histogram <-
+        ggplot2::ggplot(data.frame(x = Filter(is.finite, private$filtered_values()))) +
+        ggplot2::geom_histogram(
+          ggplot2::aes(x = x),
+          bins = 100,
+          fill = grDevices::rgb(211 / 255, 211 / 255, 211 / 255),
+          color = grDevices::rgb(211 / 255, 211 / 255, 211 / 255)
```

Comment:
I'm not sure hard coding these colors is a good idea. If I change my theme via `bslib` for example, I want all the colors of the app to match that theme. We may be able to get the theme colors via [`bslib::bs_get_variables()`](https://rstudio.github.io/bslib/reference/bs_get_variables.html).

Although, this may be better handled as part of a broader effort to [align styling across all app components](https://github.com/insightsengineering/teal/issues/735).

Reply:
yes I think this makes sense - there's a lot going on here so I'll add a note in the issue you mention